src/router.js

```diff
@@ -35,6 +36,38 @@ const logoutAndRedirect = async (to, from, next) => {
   return next('/login');
 };
 
+const checkAndSubscribeToLobby = async (to, from) => {
+  const gameStore = useGameStore();
+  const authStore = useAuthStore();
+  const gameId = parseInt(to.params.gameId);
+  
+  try {
+    if (Number.isNaN(gameId) || !Number.isFinite(gameId)) {
+      throw new Error('Invalid Lobby Number');
+    }
+
+    if (!authStore.authenticated) {
+      return {path: `/login/${gameId}` };
+    }
+
+    if (gameStore.players.some(({username}) => username === authStore.username)) {
+      return true;
+    }
+      
+    await gameStore.requestSubscribe(gameId);
+    return true;
+  }
+  catch (err) {
+    if (from.name === 'Home') {
+      throw new Error(err);
```

Comment:
Does throwing here get caught in the component that calls .push()?

Reply:
Yup